core/src/preflight/hekla.rs

```diff
@@ -0,0 +1,449 @@
+use crate::{
+    interfaces::{RaikoError, RaikoResult},
+    preflight::{block_time_to_block_slot, get_blob_data},
+    provider::{db::ProviderDb, rpc::RpcBlockDataProvider, BlockDataProvider},
+    require,
+};
+pub use alloy_primitives::*;
+use alloy_provider::{Provider, ReqwestProvider};
+use alloy_rpc_types::{Filter, Transaction as AlloyRpcTransaction};
+use alloy_sol_types::{SolCall, SolEvent};
+use anyhow::{anyhow, bail, ensure, Result};
+use kzg_traits::{
+    eip_4844::{blob_to_kzg_commitment_rust, Blob},
+    G1,
+};
+use raiko_lib::{
+    builder::{OptimisticDatabase, RethBlockBuilder},
+    clear_line,
+    consts::ChainSpec,
+    inplace_print,
+    input::{
+        proposeBlockCall, BlobProofType, BlockProposed, BlockProposedFork, GuestInput,
+        TaikoGuestInput, TaikoProverData,
+    },
+    primitives::{
+        eip4844::{self, commitment_to_version_hash, KZG_SETTINGS},
+        mpt::proofs_to_tries,
+    },
+    Measurement,
+};
+use reth_evm_ethereum::taiko::decode_anchor;
+use reth_primitives::Block;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use tracing::{debug, error, info, warn};
+
+pub async fn preflight<BDP: BlockDataProvider>(
```

Comment:
Seems a bit much to have a completely separate implementation just between HFs? Can't just have some ifs in the code for that stuff that actually changes?

Reply:
Yes, should be, splitting now for fast dev and will (as plan hopefully) remove the code dup by merging them back later.